portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.js

```diff
@@ -77,14 +85,23 @@ const renderPageWithError = (req, res, errorSummary, validationError) => {
 };
 
 const postUtilisationReportUpload = async (req, res) => {
-  const { user } = req.session;
+  const { user, utilisationReport } = req.session;
   try {
     const { uploadErrorSummary, uploadValidationError } = getUploadErrors(req, res);
     if (uploadValidationError || uploadErrorSummary) {
       return renderPageWithError(req, res, uploadErrorSummary, uploadValidationError);
     }
 
     // File is valid so we can start processing and validating its data
+    const { reportPeriod } = utilisationReport;
+    const filename = req.file.originalname;
+    const { filenameError } = validateFilenameContainsReportPeriod(filename, reportPeriod);
+    if (filenameError) {
+      const extractDataErrorSummary = [{ text: filenameError, href: '#utilisation-report-file-upload' }];
+      const extractDataError = { text: filenameError };
```

Comment:
Discussed: filename error logic has been added to `getUploadErrors`, `renderPageWithError` needs to be able to render two separate possible templates and the extraction is being left alone for now (unless we decide otherwise in later discussion)